src/msg/producer/config/writer.go

```diff
@@ -104,6 +103,12 @@ type WriterConfiguration struct {
 	Decoder                           *proto.Configuration           `yaml:"decoder"`
 	Connection                        *ConnectionConfiguration       `yaml:"connection"`
 
+	// StaticMessageRetry configs a static message retry policy.
+	StaticMessageRetry *StaticMessageRetryConfiguration `yaml:"staticMessageRetry"`
+	// MessageRetry configs a argorithmic retry policy.
+	// This takes precedence over the static retry config.
```

Comment:
Honestly, it's probably best to just error if both are set as folks may miss this comment.